server/src/main/java/com/openelements/benchscape/server/store/repositories/EnvironmentRepository.java

```diff
@@ -19,13 +19,73 @@ default List<EnvironmentEntity> findAllByQuery(@NonNull final EnvironmentQuery e
         return findAll(createSpecificationForQuery(environmentQuery));
     }
 
-    @NonNull
-    private static Specification<EnvironmentEntity> createSpecificationForQuery(
-            @NonNull final EnvironmentQuery environmentQuery) {
+    private static Specification<EnvironmentEntity> createSpecificationForQuery(@NonNull final EnvironmentQuery environmentQuery) {
         return (root, criteriaQuery, criteriaBuilder) -> {
             List<Predicate> predicates = new ArrayList<>();
+
+            final String SYSTEM_ARCH_QUERY_FIELD = "systemArch";
+            final String GIT_ORIGIN_URL_QUERY_FIELD = "gitOriginUrl";
+            final String GIT_BRANCH_QUERY_FIELD = "gitBranch";
+            final String SYSTEM_PROCESSORS_QUERY_FIELD = "systemProcessors";
+            final String SYSTEM_PROCESSORS_MIN_QUERY_FIELD = "systemProcessorsMin";
+            final String SYSTEM_PROCESSORS_MAX_QUERY_FIELD = "systemProcessorsMax";
+            final String SYSTEM_MEMORY_QUERY_FIELD = "systemMemory";
+            final String SYSTEM_MEMORY_MIN_QUERY_FIELD = "systemMemoryMin";
+            final String SYSTEM_MEMORY_MAX_QUERY_FIELD = "systemMemoryMax";
+            final String OS_NAME_QUERY_FIELD = "osName";
+            final String OS_VERSION_QUERY_FIELD = "osVersion";
+            final String JVM_VERSION_QUERY_FIELD = "jvmVersion";
+            final String JVM_NAME_QUERY_FIELD = "jvmName";
+            final String OS_FAMILY_QUERY_FIELD = "osFamily";
+            final String JMH_VERSION_QUERY_FIELD = "jmhVersion";
+
             if (environmentQuery.systemArch() != null) {
-                predicates.add(criteriaBuilder.equal(root.get("systemArch"), environmentQuery.systemArch()));
+                predicates.add(criteriaBuilder.equal(root.get(SYSTEM_ARCH_QUERY_FIELD), environmentQuery.systemArch()));
+            }
+            if (environmentQuery.gitOriginUrl() != null) {
+                predicates.add(criteriaBuilder.equal(root.get(GIT_ORIGIN_URL_QUERY_FIELD), environmentQuery.gitOriginUrl()));
+            }
+            if (environmentQuery.gitBranch() != null) {
+                predicates.add(criteriaBuilder.equal(root.get(GIT_BRANCH_QUERY_FIELD), environmentQuery.gitBranch()));
+            }
+            if (environmentQuery.systemProcessors() != null) {
+                predicates.add(criteriaBuilder.equal(root.get(SYSTEM_PROCESSORS_QUERY_FIELD), environmentQuery.systemProcessors()));
+            }
+            if (environmentQuery.systemProcessorsMin() != null) {
+                predicates.add(criteriaBuilder.equal(root.get(SYSTEM_PROCESSORS_MIN_QUERY_FIELD), environmentQuery.systemProcessorsMin()));
+            }
+            if (environmentQuery.systemProcessorsMax() != null) {
+                predicates.add(criteriaBuilder.equal(root.get(SYSTEM_PROCESSORS_MAX_QUERY_FIELD), environmentQuery.systemProcessorsMax()));
+            }
+            if (environmentQuery.systemMemory() != null) {
```

Comment:
In DB we store a long value (`Long systemMemory` in class `EnvironmentEntity`) for the memory. That is defined in bytes. Based on that the given behaviour is wrong.

I assume it will be like that:


```
 if (environmentQuery.systemMemory() != null) {
predicates.add(criteriaBuilder.equal(root.get(SYSTEM_MEMORY_QUERY_FIELD), environmentQuery.systemMemory().toBytes()));
 }
```

Reply:
`SystemMemory.toBytes()` returns the value of the memory in bytes (independent of the current unit)